### PR TITLE
Docs additions for WebGPU updates up to Chrome 115

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

There are a few changes in Chrome's WebGPU implementation between 113 and 115. See @beaufortfrancois 's update articles at https://developer.chrome.com/tags/new-in-webgpu/. This PR makes docs additions to cover the following:

- https://developer.chrome.com/blog/new-in-webgpu-115/#unset-vertex-buffer
- https://developer.chrome.com/blog/new-in-webgpu-115/#get-discrete-gpu-by-default-on-ac-power
- https://developer.chrome.com/blog/new-in-webgpu-114/#getcurrenttexture-on-unconfigured-canvas-throws-invalidstateerror
- I also removed the `async` function scaffolding from the `wgslLanguageFeatures` code snippet, as it wasn't needed.

### CODE REVIEW

File: files/en-us/web/api/gpu/requestadapter/index.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ requestAdapter(options)
 
         This hint's primary purpose is to influence which GPU is used in a multi-GPU system. For instance, some laptops have a low-power integrated GPU and a high-performance discrete GPU. Different factors may affect which adapter is returned including battery status, attached displays, or removable GPUs.
 
+        > **Note:** On dual-GPU macOS devices, if `requestAdapter()` is called without a `powerPreference` option, the high-performance discrete GPU is returned when the user's device is on AC power. Otherwise, the low-power integrated GPU is returned.
```
Comment (beaufortfrancois) [inline files/en-us/web/api/gpu/requestadapter/index.md:41]: Maybe note that this is Chrome-only behavior.

**Review by beaufortfrancois — COMMENTED** [review]
LGTM with nits

File: files/en-us/web/api/gpu/requestadapter/index.md (hunk 1 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/gpu/requestadapter/index.md:41]: Note updated to mention this.

File: files/en-us/web/api/gpurenderbundleencoder/setvertexbuffer/index.md [hunk 2]
```diff
@@ -73,6 +75,16 @@ function recordRenderPass(
 
 The above snippet is taken from the WebGPU Samples [Animometer example](https://webgpu.github.io/webgpu-samples/samples/animometer).
 
+### Unset vertex buffer
+
+```js
+// Set vertex buffer in slot 0
+passEncoder.setVertexBuffer(0, myVertexBuffer);
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/api/gpurenderbundleencoder/setvertexbuffer/index.md:82]: Good call; updated.

File: files/en-us/web/api/gpurenderpassencoder/setvertexbuffer/index.md [hunk 3]
```diff
@@ -84,6 +86,16 @@ device.queue.submit([commandEncoder.finish()]);
 // ...
 ```
 
+### Unset vertex buffer
+
+```js
+// Set vertex buffer in slot 0
+passEncoder.setVertexBuffer(0, myVertexBuffer);
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/api/gpurenderpassencoder/setvertexbuffer/index.md:93]: And the same here too

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**